internal/servers/controller/testing.go

```diff
@@ -629,3 +638,71 @@ func (tc *TestController) AddClusterControllerMember(t *testing.T, opts *TestCon
 	}
 	return NewTestController(t, nextOpts)
 }
+
+// WaitForNextWorkerStatusUpdate waits for the next status check from a worker to
+// come in. If it does not come in within the default status grace
+// period, this function returns an error.
+func (tc *TestController) WaitForNextWorkerStatusUpdate(workerId string) error {
+	tc.Logger().Debug("waiting for next status report from worker", "worker", workerId)
+
+	if err := tc.waitForNextWorkerStatusUpdate(workerId); err != nil {
+		tc.Logger().Error("error waiting for next status report from worker", "worker", workerId, "err", err)
+		return err
+	}
+
+	tc.Logger().Debug("waiting for next status report from worker received successfully", "worker", workerId)
+	return nil
+}
+
+func (tc *TestController) waitForNextWorkerStatusUpdate(workerId string) error {
+	waitStatusStart := time.Now()
+	ctx, cancel := context.WithTimeout(tc.ctx, tc.b.StatusGracePeriodDuration)
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+
+		case <-time.After(time.Second):
+			// pass
+		}
+
+		var waitStatusCurrent time.Time
+		var err error
+		tc.Controller().WorkerStatusUpdateTimes().Range(func(k, v interface{}) bool {
+			if k == nil || v == nil {
+				err = fmt.Errorf("nil key or value on entry: key=%#v value=%#v", k, v)
+				return false
```

Comment:
Don't you want to return `true` in all of these cases? Finding a bad value is obviously a bad thing, but it doesn't mean that the value for the worker ID passed in is bad. You could have perfectly legitimate data for the worker ID we care about but be ignoring it by cutting out early on an error for some other value.

Reply:
This is for type asserting that the key is a string and the value is a `time.Time`, just basically validating the data integrity in the `sync.Map`. It's more about ensuring that if there are unexpected values in `WorkerStatusUpdateTimes`, it probably means that there is a problem somewhere else and we should probably investigate it more.

I think the other thing we could do is just type assert without validation, given that this function is only designed for testing currently. Will leave for now, but let me know your thoughts.